test/tools/dshell_prebuilt/dshell_prebuilt.d

```diff
@@ -0,0 +1,313 @@
+            // Use loop to workaround issue in windows with removing
+            // executables after running then
+            for (int sleepMsecs = 10; ; sleepMsecs *= 2)
+            {
+                try {
+                    std.file.remove(expanded);
+                    break;
+                } catch (Exception e) {
+                    if (sleepMsecs >= 3000)
+                        throw e;
+                    Thread.sleep(dur!"msecs"(sleepMsecs));
+                }
+            }
+        }
+    }
+}
+
+/// Make all parent directories needed to create the given `filename`
+void mkdirFor(string filename)
+{
+    auto dir = dirName(filename);
+    //writefln("mkdirFor '%s' dir=%s", filename, dir);
+    if (!exists(dir))
+    {
+        writefln("[INFO] mkdir -p '%s'", dir);
+        mkdirRecurse(dir);
+    }
+}
+
+/// Create a file with the given `filename` and `contents`.
+void createFile(string filename, string contents)
+{
+    mkdirFor(filename);
+    auto file = File(filename, "w");
+    file.write(contents);
+}
+
+/// Append `contents` to the file named with the given `filename`.
+void appendFile(string filename, string contents)
+{
+    mkdirFor(filename);
+    auto file = File(filename, "a");
+    file.write(contents);
+}
+
+/**
+Run the given command. The `tryRun` variants return the exit code, whereas the `run` variants
+will assert on a non-zero exit code.
+*/
+auto tryRun(scope const(char[])[] args, File stdout = std.stdio.stdout, string[string] env = null)
+{
+    std.stdio.stdout.write("[RUN]");
+    if (env)
+    {
+       foreach (pair; env.byKeyValue)
+       {
+           std.stdio.stdout.write(" ", pair.key, "=", pair.value);
+       }
+    }
+    std.stdio.write(" ", escapeShellCommand(args));
+    if (stdout != std.stdio.stdout)
+    {
+        std.stdio.stdout.write(" > ", stdout.name);
+    }
+    std.stdio.stdout.writeln();
+    std.stdio.stdout.flush();
+    auto proc = spawnProcess(args, stdin, stdout, std.stdio.stderr, env);
+    return wait(proc);
+}
+/// ditto
+void run(scope const(char[])[] args, File stdout = std.stdio.stdout, string[string] env = null)
+{
+    const exitCode = tryRun(args, stdout, env);
+    if (exitCode != 0)
+    {
+        writefln("Error: last command exited with code %s", exitCode);
+        assert(0, "last command failed");
+    }
+}
+/// ditto
+void run(string cmd, File stdout = std.stdio.stdout, string[string] env = null)
+{
+    // TODO: option to disable this?
+    if (SEP != "/")
+        cmd = cmd.replace("/", SEP);
+    run(parseCommand(cmd), stdout, env);
+}
+
+/**
+Parse the given string `s` as a command.  Performs BASH-like variable expansion.
+*/
+string[] parseCommand(string s)
+{
+    auto rawArgs = s.split();
+    auto args = appender!(string[])();
+    foreach (rawArg; rawArgs)
+    {
+        args.put(expand(rawArg));
+    }
+    return args.data;
+}
+
+/// Expand the given string using BASH-like variable expansion.
+string expand(const(char)[] s)
```

Comment:
Can we name this differently, s.t. it's not confused with the tuples expand?

Reply:
Renamed it to `shellExpand`.